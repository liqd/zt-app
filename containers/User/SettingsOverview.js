import React from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'

import { Header } from '../../components/Header'
import { ListContainer, ListLink } from '../../components/List'

export const SettingsOverview = (props) => {

  // Pass fetched data forward
  const toProfileSettings = () => {
    props.navigation.navigate('SettingsProfile', {
      userId: userId,
      userName: userName,
      userImage: userImage
    })
  }

  // Pass updated info back
  const toProfileScreen = () => {
    props.navigation.navigate('ProfileScreen', {
      userId: userId,
      userName: userName,
      userImage: userImage
    })
  }

  const {userId, userName, userImage} = props.route.params

  return (
    <SafeAreaView
    >
      <Header
        transparent={true}
        handleCustomBack={toProfileScreen}
      />
      <ListContainer
        title='Profile'>
        <ListLink
          linkTitle='Edit Profile'
          onPress={toProfileSettings}/>
        <ListLink
          linkTitle='Dummy Settings'
          onPress={toProfileSettings}/>
      </ListContainer>
    </SafeAreaView>
  )
}
