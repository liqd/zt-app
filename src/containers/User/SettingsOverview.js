import React from 'react'
import { useTranslation } from 'react-i18next'
import { SafeAreaView } from 'react-native-safe-area-context'

import { Header } from '../../components/Header'
import { ListContainer, ListLink } from '../../components/List'
import { StatusBarStyled } from '../../components/StatusBarStyled'

export const SettingsOverview = (props) => {
  const { t } = useTranslation()

  const toProfileSettings = () => {
    props.navigation.navigate('SettingsProfile')
  }

  const toProfileScreen = () => {
    props.navigation.navigate('ProfileScreen')
  }

  return (
    <SafeAreaView
    >
      <StatusBarStyled />
      <Header
        transparent={true}
        handleCustomBack={toProfileScreen}
      />
      <ListContainer
        title={('Profile')}>
        <ListLink
          linkTitle={t('Edit Profile')}
          onPress={toProfileSettings}/>
      </ListContainer>
    </SafeAreaView>
  )
}
