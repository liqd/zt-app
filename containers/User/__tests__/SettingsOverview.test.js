import React from 'react'
import { render } from '@testing-library/react-native'

import '@testing-library/jest-native/extend-expect'

import { SettingsOverview } from '../SettingsOverview.js'

test('Test SettingsOverview Snapshot', async () => {
  const { toJSON } = render(<SettingsOverview />)
  expect(toJSON()).toMatchSnapshot()
})
