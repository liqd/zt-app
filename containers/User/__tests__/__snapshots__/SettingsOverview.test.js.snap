// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SettingsOverview Snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 2,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 0,
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "#1a1a1a",
                  "fontSize": 17,
                  "marginHorizontal": 5,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro_400Regular",
          },
          Object {
            "borderBottomColor": "#d8d8d8",
            "borderBottomWidth": 1,
            "borderRadius": 0,
            "fontFamily": "SourceSansPro_600SemiBold",
            "fontSize": 17,
            "paddingLeft": 15,
            "paddingVertical": 15,
          },
        ]
      }
    >
      Profile
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomColor": "#d8d8d8",
          "borderBottomWidth": 1,
          "color": "#1a1a1a",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "fontFamily": "SourceSansPro_400Regular",
            }
          }
        >
          Edit Profile
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "SourceSansPro_400Regular",
            }
          }
        >
          <Text />
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomColor": "#d8d8d8",
          "borderBottomWidth": 1,
          "color": "#1a1a1a",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "fontFamily": "SourceSansPro_400Regular",
            }
          }
        >
          Dummy Settings
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "SourceSansPro_400Regular",
            }
          }
        >
          <Text />
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
